fauna/client/client.py

```diff
@@ -579,6 +614,98 @@ def _set_endpoint(self, endpoint):
     self._endpoint = endpoint
 
 
+class StreamIterator:
+  """A class that mixes a ContextManager and an Iterator so we can detected retryable errors."""
+
+  def __init__(self, http_client: HTTPClient, headers: Dict[str, str],
+               endpoint: str, max_attempts: int, max_backoff: int,
+               opts: StreamOptions, token: StreamToken):
```

Comment:
We don't need to block this PR on it since this is essentially an internal class, but let's file a task to clean this contract up. We can collapse the various option-type params in constructor into StreamOptions and coalesce them prior to instantiation.